### PR TITLE
mon/OSDMonitor: make mapping job behave if mon_osd_prime_pg_temp = false

### PR DESCRIPTION
**Pull request by liewegas**
Make this robust to changes online too, so that we clean up a mapping
job that was started, but only use it or start a new one if the
option is true.

Signed-off-by: Sage Weil <sage@redhat.com>

### CODE REVIEW

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -380,13 +380,15 @@ void OSDMonitor::on_active()
...
     mapping_job->abort();
   }
-  C_PrintTime *fin = new C_PrintTime(osdmap.get_epoch());
-  mapping.reset(new OSDMapMapping);
-  mapping_job = mapping->start_update(osdmap, mapper,
-				      g_conf->mon_osd_mapping_pgs_per_chunk);
-  dout(10) << __func__ << " started mapping job " << mapping_job.get()
-	   << " at " << fin->start << dendl;
-  mapping_job->set_finish_event(fin);
+  if (g_conf->mon_osd_prime_pg_temp) {
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: this fixes http://tracker.ceph.com/issues/19020 also, as src/test/erasure-code/test-erasure-code.sh sets mon_osd_prime_pg_temp to false.

could you add a "Fixes" tag in the commit message?

**Comment by liewegas** [comment]
updated!

**Comment by tchaikov** [comment]
```
ceph_test_objectstore: /home/jenkins-build/build/workspace/ceph-pull-requests/src/test/objectstore/store_test.cc:3263: virtual void SyntheticWorkloadState::C_SyntheticOnClone::finish(int): Assertion `bl_eq(state->contents[noid].data, r2)' failed.
*** Caught signal (Aborted) **
 in thread 7f2c7cc65700 thread_name:fn_anonymous
 ceph version 12.0.0-662-g98f2db1 (98f2db15b9bbf1267f7737ca966f2f5c4e1523a5)
 1: (()+0x46df27) [0x7f2c8aebff27]
 2: (()+0x10330) [0x7f2c81b3f330]
 3: (gsignal()+0x37) [0x7f2c8017dc37]
 4: (abort()+0x148) [0x7f2c80181028]
 5: (()+0x2fbf6) [0x7f2c80176bf6]
 6: (()+0x2fca2) [0x7f2c80176ca2]
 7: (SyntheticWorkloadState::C_SyntheticOnClone::finish(int)+0x365) [0x7f2c8ac9e625]
 8: (Context::complete(int)+0x9) [0x7f2c8ac79ea9]
 9: (Finisher::finisher_thread_entry()+0x1de) [0x7f2c81ff4a0e]
 10: (()+0x8184) [0x7f2c81b37184]
 11: (clone()+0x6d) [0x7f2c8024137d]
2017-02-22 23:35:21.580891 7f2c7cc65700 -1 *** Caught signal (Aborted) **
 in thread 7f2c7cc65700 thread_name:fn_anonymous
```
retest this please.

**Comment by liewegas** [comment]
@tchaikov that failure looks unrelated. do you remember which test case it was though?

**Comment by tchaikov** [comment]
@liewegas i think it's another form of http://tracker.ceph.com/issues/17743.

**Comment by yuriw** [comment]
http://pulpito.front.sepia.ceph.com:80/yuriw-2017-02-24_16:24:00-rados-wip-yuri-testing2_2017_2_24-distro-basic-smithi/

http://pulpito.ceph.com/yuriw-2017-02-25_17:37:41-rados-wip-yuri-testing2_2017_2_24---basic-smithi/

**Pull request merged**